pygments/lexers/mcfunction.py

```diff
@@ -0,0 +1,326 @@
+        ],
+    }
+
+
+class MCFunctionLexer(RegexLexer):
+    """Lexer for the mcfunction scripting language used in Minecraft
+    
+
+    Modelled somewhat after the `GitHub mcfunction grammar <https://github.com/Arcensoth/language-mcfunction>`_.
+    .. versionadded:: 2.12.0
+    """
+
+    name = "MCFunction"
+    url = "https://minecraft.fandom.com/wiki/Commands"
+    aliases = ["mcfunction", "mcf"]
+    filenames = ["*.mcfunction"]
+    mimetypes = ["text/mcfunction"]
+
+    # Used to denotate the start of a block comment, borrowed from Github's mcfunction
+    _block_comment_prefix = "[>!*]"
+
+    tokens = {
+        "root": [
+            include("names"),
+            include("comments"),
+            include("literals"),
+            include("whitespace"),
+            include("property"),
+            include("operators"),
+            include("selectors"),
+        ],
+
+        "names": [
+            # The start of a command (either beginning of line OR after the run keyword)
+            #  We don't encode a list of keywords since mods, plugins, or even pre-processors
+            #  may add new commands, so we have a 'close-enough' regex which catches them.
+            (r"^(\s*)([a-z_]+)", bygroups(Whitespace, Name.Builtin)),
+            (r"(?<=run)\s+[a-z_]+", Name.Builtin),
+
+            # UUID
+            (
+                r"\b[0-9a-fA-F]+(?:-[0-9a-fA-F]+){4}\b",
+                Name.Variable,
+            ),
+            include("resource-name"),
+            # normal command names and scoreboards
+            #  there's no way to know the differences unfortuntely
+            (r"[A-Za-z_][A-Za-z0-9_.#%$]+", Keyword.Constant),
+            (r"[#%$][A-Za-z0-9_.#%$]+", Name.Variable.Magic),
+        ],
+
+        "resource-name": [
+            (
+                # resource names have to be lowercase
+                r"#?[a-z_][a-z_\.\-]*\:[a-z0-9_\.\-\/]+",
+                Name.Function,
+            ),
+            (
+                # similar to above except optional `:``
+                #  a `/` must be present "somewhere"
+                r"#?[a-z0-9_\.\-]+\/[a-z0-9_\.\-\/]+",
+                Name.Function,
+            )
+        ],
+
+        "whitespace": [
+            (r"\s+", Whitespace),
+        ],
+
+        "comments": [
+            (
+                rf"^\s*(#{_block_comment_prefix})",
+                Comment.Multiline,
+                (
+                    "comments.block",
+                    "comments.block.emphasized",
+                ),
+            ),
+            (r"#.*$", Comment.Single),
+        ],
+        "comments.block": [
+            (rf"^\s*#{_block_comment_prefix}", Comment.Multiline, "comments.block.emphasized"),
+            (r"^\s*#", Comment.Multiline, "comments.block.normal"),
+            default("#pop"),
+        ],
+        "comments.block.normal": [
+            include("comments.block.special"),
+            (r"\S+?", Comment.Multiline),
```

Comment:
That non-greedy match matches a single letter each time, which is quite inefficient. Is the `?` really necessary?